builtin/credential/ldap/backend_test.go

```diff
@@ -856,20 +877,24 @@ func TestLdapAuthBackend_ConfigUpgrade(t *testing.T) {
 
 	ctx := context.Background()
 
-	// Write in some initial config
+	cleanup, cfg := ldap.PrepareTestContainer(t, "latest")
+	defer cleanup()
 	configReq := &logical.Request{
 		Operation: logical.UpdateOperation,
 		Path:      "config",
 		Data: map[string]interface{}{
-			"url":                    "ldap://ldap.forumsys.com",
-			"userattr":               "uid",
-			"userdn":                 "dc=example,dc=com",
-			"groupdn":                "dc=example,dc=com",
-			"binddn":                 "cn=read-only-admin,dc=example,dc=com",
+			"url":                    cfg.Url,
+			"userattr":               cfg.UserAttr,
+			"userdn":                 cfg.UserDN,
+			"groupdn":                cfg.GroupDN,
+			"groupattr":              cfg.GroupAttr,
+			"binddn":                 cfg.BindDN,
+			"bindpass":               cfg.BindPassword,
 			"token_period":           "5m",
 			"token_explicit_max_ttl": "24h",
 		},
-		Storage:    storage,
+		Storage: storage,
+		// TODO remove?
```

Comment:
Is this note about not bothering to populate the connection? Don't know if it's helpful but I've found that a lot of tests here and there panic without the Connection being populated. Not sure if that's true for this one though.

Reply:
I don't remember what the TODO was about.  I'll remove the comment.